org.opent2t.sample.sensorpod.superpopular/com.wink.sensorpod/js/thingTranslator.js

```diff
@@ -0,0 +1,207 @@
+            rt: 'oic.r.sensor',
+            value: stateReader.get('brightness')
+        });
+    }
+
+    if (stateReader.containsKey('loudness')) {
+        sensors.push({
+            id: 'loudness',
+            rt: 'oic.r.sensor',
+            value: stateReader.get('loudness')
+        });
+    }
+
+    if (stateReader.containsKey('vibration')) {
+        sensors.push({
+            id: 'loudness',
+            rt: 'oic.r.sensor',
+            value: stateReader.get('vibration')
+        });
+    }
+
+    if (stateReader.containsKey('locked')) {
+        sensors.push({
+            id: 'locked',
+            rt: 'oic.r.sensor',
+            value: stateReader.get('locked')
+        });
+    }
+
+    if (stateReader.containsKey('liquid_detected')) {
+        sensors.push({
+            id: 'liquidDetected',
+            rt: 'oic.r.sensor',
+            value: stateReader.get('liquid_detected')
+        });
+    }
+
+    if (stateReader.containsKey('occupied')) {
+        sensors.push({
+            id: 'occupied',
+            rt: 'oic.r.sensor',
+            value: stateReader.get('occupied')
+        });
+    }
+
+    return sensors;
+}
+
+// Helper method to convert the device schema to the translator schema.
+function deviceSchemaToTranslatorSchema(deviceSchema) {
+
+    var stateReader = new StateReader(deviceSchema.desired_state, deviceSchema.last_reading);
+
+    return {
+        id: deviceSchema['object_type'] + '.' + deviceSchema['object_id'],
+        n: deviceSchema['name'],
+        rt: 'org.opent2t.sample.sensorpod.superpopular',
+        sensors: readSensorsArray(stateReader)
+    };
+}
+
+// Helper method to convert the translator schema to the device schema.
+function translatorSchemaToDeviceSchema(translatorSchema) {
+
+    // build the object with desired state
+    var result = { 'desired_state': {} };
+
+    if (translatorSchema.n) {
+        result['name'] = translatorSchema.n;
+    }
+
+    return result;
+}
+
+var deviceId;
+var deviceType = 'sensor_pods';
+var winkHub;
+
+// This translator class implements the 'org.opent2t.sample.sensorpod.superpopular' interface.
+class Translator {
+
+    constructor(deviceInfo) {
+        console.log('Wink Sensor Pod initializing...');
+
+        validateArgumentType(deviceInfo, "deviceInfo", "object");
+        
+        deviceId = deviceInfo.deviceInfo.id;
+        winkHub = deviceInfo.hub;
+
+        console.log('Wink Sensor Pod initializing...Done');
+    }
+
+    // exports for the entire schema object
+
+    // Queries the entire state of the sensorpod
+    // and returns an object that maps to the json schema org.opent2t.sample.sensorpod.superpopular
+    getSensorpodResURI() {
+        return winkHub.getDeviceDetailsAsync(deviceType, deviceId)
+            .then((response) => {
+                return deviceSchemaToTranslatorSchema(response.data);
+            });
+    }
+
+    // Updates the current state of the sensorpod with the contents of postPayload
```

Comment:
>state [](start = 27, length = 5)

looks like only name is allowed to be changed, which makes sense but the comment should reflect that